geo/src/algorithm/bounding_rect.rs

```diff
@@ -30,149 +28,130 @@ pub trait BoundingRect<T: CoordinateType> {
     /// assert_eq!(116.34, bounding_rect.min().y);
     /// assert_eq!(118.34, bounding_rect.max().y);
     /// ```
-    fn bounding_rect(&self) -> Self::Output;
+    fn bounding_rect(&self) -> Result<Rect<T>, ()>;
```

Comment:
Hey @b4l - thanks for your interest and the PR! 

Before going into your implementation, I raised some concerns with the specification of the work over at https://github.com/georust/geo/issues/444#issuecomment-746600949. Sorry for not looking into it more closely before you started working on this.

I encourage you to engage in that discussion if you have any thoughts.